Lombiq.NodeJs.Extensions/scripts/run-pnpm-script.js

```diff
@@ -0,0 +1,77 @@
+const fs = require('fs');
+const path = require('path');
+const { execSync, exec } = require('child_process');
+
+const { handleErrorObject } = require('./handle-error');
+
+function panic(message) {
+    handleErrorObject(message);
+    process.exit(1);
+}
+
+// Check if pnpm is installed.
+try { execSync('pnpm -v'); }
+catch (_) { panic('This project requires pnpm. You can install it with the "npm install --global pnpm" command.'); }
+
+// Load command line arguments.
+const args = process.argv.splice(2);
+if (args.length < 2) panic('USAGE: node scripts/run-pnpm-script project-path script');
+
+// Initialize variables.
+const [projectPath, script] = args;
+const packageJsonPath = path.join(projectPath, 'package.json');
+
+function call(command) {
+    process.stdout.write(`Executing "${command}"...`);
+
+    return new Promise((resolve, reject) => {
+        exec(command, { }, (error, stdout, stderr) => {
+            process.stdout.write(stdout);
+            process.stderr.write(stderr);
+            (error ? reject : resolve)(error);
+        });
+    });
+}
+
+function callScriptInLibrary(scriptToExecute) {
+    return call('npm explore nodejs-extensions -- pnpm ' + scriptToExecute);
+}
+
+async function main() {
+    // Create the package.json file if it doesn't exist.
+    if (!fs.existsSync(packageJsonPath)) {
+        panic(`Couldn't find "${packageJsonPath}".`);
+    }
+
+    // Go to project directory and read the package.json file to find the scripts.
+    process.chdir(projectPath);
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf8' }));
+    const scripts = packageJson.scripts ?? { };
+
+    // Handle copy:assets which is a special case during build/compile.
+    if ((script === 'build' || script === 'compile') && Array.isArray(packageJson.nodejsExtensions?.assetsToCopy)) {
+        await callScriptInLibrary('copy:assets');
+    }
```

Comment:
How about - instead of this hack - we introduce a convention for the supported script "types" (aka prefixes) and rename `copy:assets` to `build:assets` or even `build:copyAssets`?

Reply:
I went with `build:assets`, because I had the intuition that these script names ought to be snake-case. That would be `build:copy-assets` instead of `build:copyAssets`, but I don't want to leave this hanging. Also I've kept `copy:assets` in _Lombiq.NodeJs.Extensions/package.json_ for the sake of backwards compatibility, but I've removed it from the readme to discourage future use.